pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go

```diff
@@ -748,6 +748,12 @@ func (r *Reconciler) runReconcileShootFlow(ctx context.Context, o *operation.Ope
 			SkipIf:       o.Shoot.IsWorkerless || o.Shoot.HibernationEnabled,
 			Dependencies: flow.NewTaskIDs(deployWorker),
 		})
+		deployExtensionResourcesAfterWorker = g.Add(flow.Task{
+			Name: "Deploying extension resources after workers",
+			Fn:   flow.TaskFn(botanist.DeployExtensionsAfterWorker).RetryUntilTimeout(defaultInterval, defaultTimeout),
+			SkipIf:       o.Shoot.IsWorkerless || o.Shoot.HibernationEnabled,
```

Comment:
Why is this skipped when hibernating clusters? There might be extensions that need to handle hibernation.
For extensions using `BeforeKubeAPIServer`, there are two tasks: one is skipped for hibernated clusters and one is only executed for hibernated clusters. This is done to invert the lifecycle strategy during hibernation.

In the case of `AfterWorker`, we shouldn't need this inversion. Instead, we should always execute the deploy task (same as for `AfterKubeAPIServer`).

Reply:
My though was that there is nothing that needs to be done for an extension, but you are right, there MIGHT be something, I'll update the skip condition.